applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_vmsmonolithic.py

```diff
@@ -306,10 +342,15 @@ def Initialize(self):
             self.EstimateDeltaTimeUtility = self._GetAutomaticTimeSteppingUtility()
 
         # Creating the solution strategy
-        self.conv_criteria = KratosCFD.VelPrCriteria(self.settings["relative_velocity_tolerance"].GetDouble(),
-                                                     self.settings["absolute_velocity_tolerance"].GetDouble(),
-                                                     self.settings["relative_pressure_tolerance"].GetDouble(),
-                                                     self.settings["absolute_pressure_tolerance"].GetDouble())
+
+        if self.settings["time_scheme"].GetString() == "steady":
+            self.conv_criteria = KratosMultiphysics.ResidualCriteria(self.settings["relative_velocity_tolerance"].GetDouble(),
+                                                                     self.settings["absolute_velocity_tolerance"].GetDouble())
```

Comment:
Why does the steady scheme need a different criteria? (And why is it taking the velocity values?)
If possible, I would leave the choice of convergence criteria independent of the time scheme: residual can/should work with bossak too, and vel-pr with steady.

Reply:
The issue was with the steady scheme. In order to steady scheme to work with VelPr criteria, one has to give really hard tolerances (at least for the tests which i was using). 

I agree using using only the "velocity tolerances" is not ideal. The residual crieterian can only have one relative and one absolute tolerance, and i cannot get one value for both velocity and pressure without changing the input defaults for json. thats why I used only the velocity tolerances.

If we do it in a clean way by having a factory for convergence criterions, we can get rid of this method. but it will need to change the input defaults of the json.

Reply:
We should go for a factory approach to set the convergence criteria at some point, but that would be a deeper change (due to the change of defaults on all fluid solvers). I agree that this is better left for a future PR.